pilot/pkg/model/service.go

```diff
@@ -78,6 +78,10 @@ type Service struct {
 	// Do not access directly. Use GetAddressForProxy
 	DefaultAddress string `json:"defaultAddress,omitempty"`
 
+	// ExtraDefaultAddresses specifies the extra service addresses of the load balancer.
+	// This field can be used by dual stack service to store other IP family addresses.
+	// Do not access directly. Use GetExtraSvcAddresses
+	ExtraSvcAddresses []string `json:"extraSvcAddresses,omitempty"`
```

Comment:
`ClusterVIPs` already holds a map of `Cluster->[]Address`. What does it mean to have a singular `ExtraSvcAddresses` when we can have multiple clusters?

It seems like `ClusterVIPs` _already_ holds the correct data. The issue is more that `GetAddressForProxy` ignores all of them (`return addresses[0]`). It seems like we could have an alternative method `GetAddressesForProxy` that returns all of them...? WDYT?

Long term we can remove GetAddressForProxy, but its probably hard to update every single call site at once to support multiple

Reply:
Hi @howardjohn, yeah, it's a great suggestion,  `ClusterVIPs` have all necessary addresses. BTW, there is an existing function named `getAllAddresses()` can returns all addresses of the service, so there are 2 options:

- [ ] 1. I can change the name as `GetAllAddresses()`
- [ ] 2. Create a new function named `GetAddressesForProxy` and pass the proxy node as its parameter

Which is the better?

Reply:
Hi @howardjohn, I think the `ExtraSvcAddresses` is needed for the Istio service converted from service entry, especially in dual stack cluster. Because the `ClusterVIPs` of Istio service is empty in this situation.